crutest/src/main.rs

```diff
@@ -846,8 +844,79 @@ async fn main() -> Result<()> {
         pr = None;
     }
 
-    let _join_handle = up_main(crucible_opts, opt.gen, None, io, pr)?;
-    println!("Crucible runtime is spawned");
+    // We need to build a Volume for all the tests to use.
+    // If we have received a VCR as input, we can use that.  Otherwise we
+    // have to construct one by asking our downstairs for information that
+    // we need up front.  This will work as long as one of the downstairs
+    // is up already.  If we have a test that requires no downstairs to be
+    // running on startup, then we need to provide a VCR up front.
+    let guest = {
+        if let Some(vcr_file) = opt.vcr_file {
+            let vcr: VolumeConstructionRequest = match read_json(&vcr_file) {
+                Ok(vcr) => vcr,
+                Err(e) => {
+                    bail!("Error {:?} reading VCR from {:?}", e, vcr_file)
+                }
+            };
+            let volume =
+                Volume::construct(vcr, pr, guest_logger).await.unwrap();
+            Arc::new(volume)
+        } else {
+            // We were not provided a VCR, so, we have to make one by using
```

Comment:
Vague architectural question: do we need to get this from the repair port, or could we get it by building a `Guest` instead (then use that `Guest` to build a `Volume`)?

Reply:
Maybe.. we could build it backwards like that.
There exists a `Volume`:

```
    // Create a simple Volume from a single BlockIO                             
    pub async fn from_block_io(                                                 
        block_io: Arc<dyn BlockIO + Sync + Send>,                               
        log: Logger,
    ) -> Result<Volume, CrucibleError> {                                       
```

But, I would have to play around with it to be sure it does what we need.

We could also possibly just make a `Guest` here instead of a `Volume`, though I was not sure if I could have the
```
let guest = {
```
Return either a `Volume` or a `Guest`.
Let me see if the `from_block_io` works like I expect.

Reply:
Ah, so, the problem is that the guest layer can't return a valid `get_block_size()` until the
upstairs is activated.  This means the `from_block_io()` volume creation attempt fails.
Even if `get_block_size()` worked, we also need `total_size()` to work as well.

So, for a Volume, we have to know this information ahead of time.  We could require it be passed in, or
use the method in the PR to scrape it from a downstairs.

Reply:
Okay, seems fine in that case